waspc/data/Generator/templates/server/src/crud/_helpers.ts

```diff
@@ -0,0 +1,77 @@
+{{={= =}=}}
+import {
+  {=# isAuthEnabled =}
+  type {= userEntityUpper =},
+  {=/ isAuthEnabled =}
+} from '../entities/index.js'
+// TODO(miho): replace with proper imports once we support type imports
+import { PrismaArgs, type RouteInputs } from '../universal/crud/{= crud.name =}.js'
+import { Expand, PartialBy } from '../universal/types.js'
+
+export type { PrismaArgs, RouteInputs }
+
+{=# isAuthEnabled =}
+type UserEntity = {= userEntityUpper =}
+{=/ isAuthEnabled =}
+{=^ isAuthEnabled =}
+type UserEntity = any
+{=/ isAuthEnabled =}
+
+namespace OverridesReturnValues {
+  // We provide the default 'where' property for Get operation
+  export type Get = Expand<PartialBy<PrismaArgs.Get, 'where'>>
+  export type GetAll = PrismaArgs.GetAll
+  // We provide the default 'data' property for Create operation
+  export type Create = Expand<PartialBy<PrismaArgs.Create, 'data'>>
+  // We provide the default 'where' and 'data' properties for Update operation
+  export type Update = Expand<PartialBy<PrismaArgs.Update, 'where' | 'data'>>
+  // We provide the default 'where' property for Delete operation
+  export type Delete = Expand<PartialBy<PrismaArgs.Delete, 'where'>>
+}
+
+type Overrides = {
+    {=# crud.operations.Get =}
+    {=# isPublic =}
+    Get?: (args: RouteInputs.Get, user?: UserEntity) => OverridesReturnValues.Get
+    {=/ isPublic =}
+    {=^ isPublic =}
+    Get?: (args: RouteInputs.Get, user: UserEntity) => OverridesReturnValues.Get
+    {=/ isPublic =}
+    {=/ crud.operations.Get =}
+    {=# crud.operations.GetAll =}
+    {=# isPublic =}
+    GetAll?: (args: RouteInputs.GetAll, user?: UserEntity) => OverridesReturnValues.GetAll
+    {=/ isPublic =}
+    {=^ isPublic =}
+    GetAll?: (args: RouteInputs.GetAll, user: UserEntity) => OverridesReturnValues.GetAll
+    {=/ isPublic =}
+    {=/ crud.operations.GetAll =}
+    {=# crud.operations.Create =}
+    {=# isPublic =}
+    Create?: (args: RouteInputs.Create, user?: UserEntity) => OverridesReturnValues.Create
+    {=/ isPublic =}
+    {=^ isPublic =}
+    Create?: (args: RouteInputs.Create, user: UserEntity) => OverridesReturnValues.Create
+    {=/ isPublic =}
+    {=/ crud.operations.Create =}
+    {=# crud.operations.Update =}
+    {=# isPublic =}
+    Update?: (args: RouteInputs.Update, user?: UserEntity) => OverridesReturnValues.Update
+    {=/ isPublic =}
+    {=^ isPublic =}
+    Update?: (args: RouteInputs.Update, user: UserEntity) => OverridesReturnValues.Update
+    {=/ isPublic =}
+    {=/ crud.operations.Update =}
+    {=# crud.operations.Delete =}
+    {=# isPublic =}
+    Delete?: (args: RouteInputs.Delete, user?: UserEntity) => OverridesReturnValues.Delete
+    {=/ isPublic =}
+    {=^ isPublic =}
+    Delete?: (args: RouteInputs.Delete, user: UserEntity) => OverridesReturnValues.Delete
+    {=/ isPublic =}
+    {=/ crud.operations.Delete =}
+}
+
+export function createCrudOverrides(overrides: Overrides): Overrides {
```

Comment:
This is a good point, but if hook works much better, maybe we can go with it, as an experiment? I would want us to do stuff in worse way just because we did it like that before :D. And this is not an argument to remove Haskell and DSL also :D.

Reply:
@infomiho what are your thoughts on this?

Reply:
That's an interesting one, since we changed the overrides approach to be more like what have for operations, the concept of `createCrudOverrides` doesn't exist anymore.

I could implement `createGetOverride`, `createGetAllOverride` etc. functions and expose it for each crud? I'm on the fence but it might be easier for users to use that mechanism?

Reply:
> But you now implemented these via DSL, didn't you?

The discussion at hand doesn't have to do anything with DSL but rather the way we provide Typescript types when defining the override implementations.

First approach is to export types like `GetQuery` and then users import them and use like this:
```ts
import { GetQuery } from '@wasp/crud/tasks'

const getOverride: GetQuery<Input, Output> = (args, context) => {
  // args and context are typed here
}
```

Second approach is to offer an `id` function that provides types like this:
```ts
import { createGetOverride } from '@wasp/crud/tasks'

export const getOverride = createGetOverride<Input, Output>((args, context) => {
  // args and context are typed here
}) 
```

So, it's a matter of trying out a new approach or doing what's familiar 🤷  Right now, we are doing what's familiar.

Reply:
Got it! Ok, will let you judge what you think it the best! It does make sense to keep things consistent though.